instrumentation/opentelemetry-instrumentation-starlette/src/opentelemetry/instrumentation/starlette/__init__.py

```diff
@@ -176,10 +176,12 @@ def client_response_hook(span: Span, scope: dict[str, Any], message: dict[str, A
 
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Any, Collection, cast
+from functools import partial
+from typing import TYPE_CHECKING, Any, Collection
 
 from starlette import applications
 from starlette.routing import Match
```

Comment:
So if we don't have an ordering issue with other instrumentations then the value of switching to wrapt would be on removing these top level imports and push them at a later time or move them under type checking. Do you see this achievable?